### PR TITLE
Update to python3

### PR DESCRIPTION
**Pull request by jiaqi216**
### New Features
- ran 2to3, uses python3 syntax
- added wool  and ran black for formating

Note: I removed [gdcdatamodel/models/](https://github.com/uc-cdis/gdcdatamodel/pull/23/files#diff-a084b794bc0759e7a6b77810e01874f2L8) from .gitignore. Not sure what the motivation was for having it in there but needed to update the files to be python3 compatible

### Breaking Changes
Moves to python3, drops py2 support

### Dependency updates
* dictionaryutils on py3
* gdcdictionary on py3
* psqlgraph on py3

### CODE REVIEW

File: gdcdatamodel/models/__init__.py [hunk 1]
```diff
@@ -115,8 +114,8 @@ def _recursive_get_links(links, result):
 def types_from_str(types):
     return [a for type_ in types for a in {
         'string': [str],
-        'number': [float, int, long],
-        'integer': [int, long],
+        'number': [float, int, int],
+        'integer': [int, int],
```
Comment (paulineribeyre) [inline gdcdatamodel/models/__init__.py]: remove double `int`? :p

File: gdcdatamodel/models/__init__.py [hunk 2]
```diff
@@ -380,7 +379,7 @@ def generate_edge_tablename(src_label, label, dst_label):
         oldname = tablename
         logger.debug('Edge tablename {} too long, shortening'.format(oldname))
         tablename = 'edge_{}_{}'.format(
-            str(hashlib.md5(tablename).hexdigest())[:8],
+            str(hashlib.md5(tablename.encode("latin1")).hexdigest())[:8],
```
Comment (paulineribeyre) [inline gdcdatamodel/models/__init__.py]: `uft-8`?

File: gdcdatamodel/models/indexes.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def index_name(cls, description):
         oldname = index_name
         logger.debug('Edge tablename {} too long, shortening'.format(oldname))
         name = 'index_{}_{}_{}'.format(
-            str(hashlib.md5(cls.__tablename__).hexdigest())[:8],
+            str(hashlib.md5(cls.__tablename__.encode('latin1')).hexdigest())[:8],
```
Comment (paulineribeyre) [inline gdcdatamodel/models/indexes.py]: `utf-8`?

File: gdcdatamodel/models/indexes.py [hunk 4]
```diff
@@ -103,4 +103,4 @@ def get_secondary_key_indexes(cls):
 def cls_add_indexes(cls, indexes):
     """Add indexes to given class"""
 
-    map(cls.__table__.append_constraint, indexes)
+    list(map(cls.__table__.append_constraint, indexes))
```
Comment (paulineribeyre) [inline gdcdatamodel/models/indexes.py]: no need for list

File: setup.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (paulineribeyre) [inline setup.py]: black will probably complain about this

File: setup.py [hunk 6]
```diff
@@ -25,8 +27,7 @@ def get_version():
         'pytz>=2016.4',
         'graphviz~=0.4',
         'jsonschema~=2.5',
-        'psqlgraph~=2.0',
-        'dictionaryutils>=1.2.0,<3.0.0',
+        'psqlgraph',
```
Comment (paulineribeyre) [inline setup.py]: why did you remove the pin?

File: Pipfile [hunk 7]
```diff
@@ -7,10 +7,22 @@ verify_ssl = true
...
+dictionaryutils = {editable = true,git = "https://github.com/uc-cdis/dictionaryutils.git",ref = "chore/py3"}
+gdcdictionary = {editable = true,git = "https://github.com/NCI-GDC/gdcdictionary.git",ref = "release/py3"}
+more-itertools = "==7.2.0"
+zipp = "==0.6.0"
+importlib-metadata = "==0.23"
+pyrsistent = "==0.15.4"
+pytz = "*"
+psycopg2-binary = "==2.8.2"
+sqlalchemy = "==1.3.3"
+rstr = "==2.2.6"
```
Comment (paulineribeyre) [inline Pipfile]: how come you had to add all of these dependencies?

Comment (jiaqi216) [reply Pipfile]: pipenv kept throwing an error on the vcs depedencies so I ran `pipenv install -e <vcs dependecy> ` to autogenerate the Pipfile. It should be okay to have them here since they were in the pipfile.lock to begin with. Although, eventually once we merge all the py3 branches I'd want to move away from vcs dependencies all together so this file will likely be edited at that time to remove all these packages

File: gdcdatamodel/models/__init__.py (hunk 2 above)
Comment (jiaqi216) [reply gdcdatamodel/models/__init__.py]: md5 hash will differ based on how the string is encoded. `latin1` was default in python2 but looks like we specified encoding in `utf-8` anyway https://github.com/uc-cdis/gdcdatamodel/pull/23/files/f41902a05b8bccd435fd7cc6df7518516a853e3f#diff-1ba9e31512b7e1134d72fe3ea8f7d452L1 so` utf-8` should be fine. How much do we care that these hashes are consistent anyway? If it's super important, we should probably try to come up with some test cases for this.

File: setup.py (hunk 6 above)
Comment (jiaqi216) [reply setup.py]: I think because running python setup.py install would overwrite the pipenv installed version of psqlgraph with the one from pypi. We should probably pin it once we release py3 psqlgraph to pypi.
